.goreleaser.yml

```diff
@@ -46,3 +45,66 @@ dockers:
   - "--label=org.opencontainers.image.version={{.Version}}"
   - "--label=org.opencontainers.image.source={{.GitURL}}"
   - "--platform=linux/amd64"
+- image_templates:
+  - 'skibish/ddns:{{ .Tag }}-arm64'
+  dockerfile: Dockerfile
```

Comment:
Looking at this second time, it seems that  following variables should be set, otherwise goreleaser will not understand which binaries to put into which image. Binary is built outside the Dockerfile and then it's just copied inside.

```
    # GOOS of the built binaries/packages that should be used.
    goos: linux

    # GOARCH of the built binaries/packages that should be used.
    goarch: amd64

    # GOARM of the built binaries/packages that should be used.
    goarm: ''

    # GOAMD64 of the built binaries/packages that should be used.
    goamd64: 'v2'
```

https://goreleaser.com/customization/docker/#customization

Reply:
I made these changes, but it appears to have broken the build when I ran `goreleaser release --snapshot --rm-dist`.  Not sure what's going on; I'll take a look again tomorrow and report back.

Here's the error I'm seeing:
```
#7 [stage-1 2/2] COPY ddns /
#7 ERROR: "/ddns" not found: not found
------
 > [stage-1 2/2] COPY ddns /:
------
error: failed to solve: failed to compute cache key: "/ddns" not found: not found
```

Reply:
I tried to build it on my side - the issue is with the Windows docker image. `Dockerfile` has an instruction to copy certificates chain so it would be possible to connect over HTTPS. It copies them from the `alpine` and it fails because there is no alpine version for windows. Additionally, Windows requires placement of certificates in some other place... My proposal is not to build the Docker image for Windows for now (`i386`), this should fix the issue.

I used the following variables for respective builds:

```yaml
goarch: 'arm64'
---
goarch: 'arm'
goarm: '6'
---
goarch: 'arm'
goarm: '7'
```

Result:

```bash
REPOSITORY     TAG           IMAGE ID       CREATED          SIZE
skibish/ddns   4.0.4-armv7   4d060406e2b9   4 minutes ago    7.87MB
skibish/ddns   4.0.4-amd64   799fa8cbb1fe   4 minutes ago    8.26MB
skibish/ddns   4.0.4-armv6   b6a4c5bfd990   4 minutes ago    7.87MB
skibish/ddns   4.0.4-arm64   e350a1b83a55   4 minutes ago    8MB
```

Reply:
Huh, that's not the behavior that I see... I didn't think the goreleaser file was building docker images for Windows, only the binaries.  So there shouldn't be a need to remove any sections for i386, at least not the docker sections.

I managed to get it to work, and was able to verify that the generated binaries were for the correct architectures.  Turns out I had a typo; you need to specify `goarm: 6`, not `goarm: v6`.  Goreleaser is apparently picky about the formatting for its yml file...  

Give the latest commit a try and see if it works for you.

Reply:
Works, let's merge and see if it will work from `main` branch :)